include/platform/mir/options/default_configuration.h

```diff
@@ -74,6 +75,47 @@ class DefaultConfiguration : public Configuration
     char const** const argv;
     std::function<void(int argc, char const* const* argv)> const unparsed_arguments_handler;
     std::shared_ptr<boost::program_options::options_description> const program_options;
+
+    template<typename Value>
+    class LibraryMap
```

Comment:
Why not use a `map` or `unordered_map`? Can't quite figure out the difference in functionality

Reply:
In the current code? It's got a slightly more pleasant interface for what we need.

As to why this exists? This is somewhat of a path-dependent thing. So, it was originally trying to be `unordered_map<SharedLibrary, ...>`, but `SharedLibrary` is non-copyable. So, I wanted `unordered_map<SharedLibrary*, ...>` with a custom equals and hash, but that was surprisingly difficult.

Now that we've gone to `SharedLibrary::Handle`, we could *probably* move back to a `std::unordered_map<SharedLibrary::Handle, ...>` with a slightly less pleasant interface (needing to call `.get_handle()` in various places).